lumen/base.py

```diff
@@ -39,13 +42,26 @@ def refs(self):
     def _get_type(cls, component_type):
         clsname = cls.__name__
         clslower = clsname.lower()
+        if component_type is None:
+            raise ValueError(f"'type' for '{clslower}' is not available.")
         if '.' in component_type:
             return resolve_module_reference(component_type, cls)
         try:
             __import__(f'lumen.{clslower}s.{component_type}')
         except Exception:
             pass
+
+        cls_types = set()
         for component in param.concrete_descendents(cls).values():
-            if getattr(component, f'{clsname.lower()}_type') == component_type:
+            cls_type = getattr(component, f'{clsname.lower()}_type')
+            if cls_type is None:
+                continue
+            cls_types.add(cls_type)
+            if cls_type == component_type:
                 return component
-        raise ValueError(f"No {clsname} for {clslower}_type '{component_type}' could be found.")
+
+        msg = f"No '{clslower}' for {clslower}_type '{component_type}' could be found."
+        matches = "', '".join(get_close_matches(component_type, cls_types))
+        if matches:
+            msg += f" Did you mean: '{matches}'?"
+        raise ValueError(msg)
```

Comment:
Should this code be in Param instead? It doesn't seem specific to Lumen.

Reply:
The `validate_parameters` could/should be implemented in Param as an option instead of "only" emitting a warning. https://github.com/holoviz/param/issues/174 

For the code in this function, in particular, I don't think it can be ported to param.

Reply:
You don't think that Param could raise a helpful exception when given an attribute name that is not a parameter, guessing the right parameter name the same as done here? I must be missing something, then.

Reply:
Yes! A helpful exception for Param would be a very nice addition. 

But that is not what this function does, which tries to match the input string `component_type` to a lumen component and return it.